### PR TITLE
Fixes https://github.com/strapi/strapi/issues/467

### PR DESCRIPTION
**Pull request by ayadav**

### CODE REVIEW

**Comment by lauriejim** [comment]
Hi @ayadav.

If you do that, plugins models will never be loaded, see https://github.com/joshtechnologygroup/strapi/blob/5d9a49256a612181dec6f731d59854a546787dd5/packages/strapi-bookshelf/lib/index.js#L397

You can update this line with your condition https://github.com/joshtechnologygroup/strapi/blob/5d9a49256a612181dec6f731d59854a546787dd5/packages/strapi-bookshelf/lib/index.js#L394 and delete this line https://github.com/joshtechnologygroup/strapi/blob/5d9a49256a612181dec6f731d59854a546787dd5/packages/strapi-bookshelf/lib/index.js#L74 cause it's useless code.

Thanks for your contribution !

**Comment by ayadav** [comment]
Hi @lauriejim,

Agreed, I have taken a slightly different approach this time. 

1. Used await to ensure all connections/models are processed before the callback.
2. Cleaned up all the other _after callback counters. 

On side note, It seems like the same issue could be present in other plugins e.g. strapi-mongoose but I have not touched them for now.

File: packages/strapi-bookshelf/lib/index.js [hunk 1]
```diff
@@ -40,12 +40,10 @@ module.exports = function(strapi) {
...
      */
 
-    initialize: cb => {
+    initialize: async cb => {
       const connections = _.pickBy(strapi.config.connections, { connector: 'strapi-bookshelf' });
 
-      const done = _.after(_.size(connections), cb);
-
-      _.forEach(connections, (connection, connectionName) => {
+      await _.forEach(connections, (connection, connectionName) => {
```
Comment (Aurelsicoko) [inline packages/strapi-bookshelf/lib/index.js]: I didn't know this syntax... It tested it and it doesn't change anything in my scripts. 

@ayadav Can you explain to me why we need to put the `await` reserved word?

Comment (ayadav) [reply packages/strapi-bookshelf/lib/index.js]: @Aurelsicoko You are right, I checked again and it actually had no impact. I have removed those now.

**Pull request merged**